core/src/main/java/com/netflix/conductor/core/execution/DeciderService.java

```diff
@@ -524,23 +524,38 @@ void checkTaskTimeout(TaskDef taskDef, Task task) {
             LOGGER.warn("Missing task definition for task:{}/{} in workflow:{}", task.getTaskId(), task.getTaskDefName(), task.getWorkflowInstanceId());
             return;
         }
-        if (task.getStatus().isTerminal() || taskDef.getTimeoutSeconds() <= 0 || task.getStartTime() <= 0) {
+        if (task.getStatus().isTerminal()) {
             return;
         }
 
-        long timeout = 1000L * taskDef.getTimeoutSeconds();
-        long now = System.currentTimeMillis();
-        long elapsedTime = now - (task.getStartTime() + ((long) task.getStartDelayInSeconds() * 1000L));
+        String reason = null;
+        final long now = System.currentTimeMillis();
+        final long startDelay = 1000L * task.getStartDelayInSeconds();
 
-        if (elapsedTime < timeout) {
-            return;
+        if (task.getStatus().equals(SCHEDULED) && taskDef.getPollTimeoutSeconds() > 0) {
```

Comment:
if `taskDef.getPollTimeoutSeconds()` is `null` (the case for all tasks defined before that functionality has been merged, and it's optional), this check is a bit weird.

Perhaps only checking it's not null is enough instead.

Reply:
I think both need to be checked. We can assume (to write down on the doc) that if pollTimeoutSeconds is set to 0, no checkTaskTimeout should be performed in this case

Reply:
Done